src/librustc_mir/monomorphize/collector.rs

```diff
@@ -195,9 +195,15 @@ use rustc_middle::ty::print::obsolete::DefPathBasedNames;
 use rustc_middle::ty::subst::{GenericArgKind, InternalSubsts};
 use rustc_middle::ty::{self, GenericParamDefKind, Instance, Ty, TyCtxt, TypeFoldable};
 use rustc_session::config::EntryFnType;
+use rustc_span::{Span, DUMMY_SP};
 use smallvec::SmallVec;
 use std::iter;
 
+struct MonoAndSource<'tcx> {
+    item: MonoItem<'tcx>,
+    source: Span,
+}
```

Comment:
This could just be a `rustc_span::Spanned<MonoItem>`. The name of this type should still have `Item` in it.